### DISCUSSION TITLE
Backend tests failing in fresh repository after editing .env

### DISCUSSION THREAD

**Discussion by dupanovic** [fastapi/full-stack-fastapi-template#1415] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
- Clone the repo
- Replace `changethis` values in `.env`
- In `~`, run: `docker compose watch`
- In `~/backend`, run:  `bash ./scripts/test.sh`
```

### Description

- Clone the repo
- In `~`, run: `docker compose watch`
- In `~/backend`, run:  `bash ./scripts/test.sh`

Result: all test cases pass if env is not edited and has default changethis.

However if I edit the .env and change the values that had `changethis` set, 15 test cases fail, and the first superuser row is deleted from the users table.

Every error comes from the post request in `~/backend/app/tests/utils/utils.py` at L23:
```python
r = client.post(f"{settings.API_V1_STR}/login/access-token", data=login_data)
print(r.json()) # {'detail': 'Incorrect email or password'}
``` 

Before running the tests, it is possible to successfully login with the email and password into the dashboard on :5173.

![image](https://github.com/user-attachments/assets/045cee7c-e1bd-4c4c-8ff1-e82fb09a40db)

### Operating System

Linux

### Operating System Details

Debian 12

### Python Version

3.11.2

**Comment by AleksDominik** [comment]
I have the same issue and I think it come from this [test ](https://github.com/fastapi/full-stack-fastapi-template/blob/master/backend/app/tests/api/routes/test_login.py#L72).
Once  you change the password on .env file it doesn't correspond to the updated password("changethis").
The dummy solution wil be to delete this test.
Or to find a better way to scope the session fixture to avoid the test modification on db.

**Reply by cubinet-code** [reply]
I just updated the data sent in this test to use the same password:
`data = {"new_password": settings.FIRST_SUPERUSER_PASSWORD, "token": token}`

**Reply by 53845714nF** [reply]
oh damn, that problem took me 2 hours. And now I find it here in the discussion. Replacing the old password with the same one defeats the purpose of the test, but I'll take it.

**Reply by padurean** [reply]
Right: the problem is that:
- The [`test_reset_password`](https://github.com/fastapi/full-stack-fastapi-template/blob/master/backend/app/tests/api/routes/test_login.py#L72) test modifies the user password in the database
- Other tests rely on the fact that the password is the one configured in [.env](https://github.com/fastapi/full-stack-fastapi-template/blob/ba1706b48994605580c3e246dfd416afb6124546/.env#L23) which is not the case anymore once the above test is runs => other tests fail with a _... KeyError: 'access_token'_ message

> [!NOTE]
> The database session is reused among tests because the [`db()` test fixture](https://github.com/fastapi/full-stack-fastapi-template/blob/ba1706b48994605580c3e246dfd416afb6124546/backend/app/tests/conftest.py#L15) here has scope `session`. Changing the scope to `function` for **all** test fixtures in [conftest.py](https://github.com/fastapi/full-stack-fastapi-template/blob/master/backend/app/tests/conftest.py) would fix the problem, but it would also cause the tests to take significantly longer to run.

> [!TIP]
> A solution would be to change the [`test_reset_password`](https://github.com/fastapi/full-stack-fastapi-template/blob/master/backend/app/tests/api/routes/test_login.py#L72) test so that it puts back the original (configured) password:
> ```python
> def test_reset_password(
>     client: TestClient, superuser_token_headers: dict[str, str], db: Session
> ) -> None:
>     def test_change_password(new_pass: str) -> None:
>         token = generate_password_reset_token(email=settings.FIRST_SUPERUSER)
>         data = {"new_password": new_pass, "token": token}
>         r = client.post(
>             f"{settings.API_V1_STR}/reset-password/",
>             headers=superuser_token_headers,
>             json=data,
>         )
>         assert r.status_code == 200
>         assert r.json() == {"message": "Password updated successfully"}
> 
>         user_query = select(User).where(User.email == settings.FIRST_SUPERUSER)
>         user = db.exec(user_query).first()
>         assert user
>         assert verify_password(data["new_password"], user.hashed_password)
> 
>     test_change_password("changethis")
>     # put back the password
>     test_change_password(settings.FIRST_SUPERUSER_PASSWORD)
> ```

**Comment by alejsdev** [comment, marked as answer]
Hello everyone, this was handled in #1499, thanks for the patience (and the report)! :smile_cat:
